src/runtime/pkg/oci/utils.go

```diff
@@ -566,6 +566,9 @@ func addHypervisorPathOverrides(ocispec specs.Spec, config *vc.SandboxConfig, ru
 		if value != "" {
 			params := vc.DeserializeParams(strings.Fields(value))
 			for _, param := range params {
+				if !config.HypervisorConfig.HugePages && strings.Contains(param.Key, "hugepage") {
+					continue
```

Comment:
Maybe writing a warning log will more helpful to let users know that they have set an invalid parameters combination.